lemminx-maven/src/main/java/org/eclipse/lemminx/maven/PluginValidator.java

```diff
@@ -82,80 +76,66 @@ public Diagnostic validateConfiguration(DiagnosticRequest diagnosticRequest) {
 			// A diagnostic was already added in validatePluginResolution()
 		}
 		if (parameters.isEmpty()) {
-			return null;
+			return;
 		}
 		if (node.isElement() && node.hasChildNodes()) {
 			for (DOMNode childNode : node.getChildren()) {
-				DiagnosticRequest childDiagnosticReq = new DiagnosticRequest(childNode,
-						diagnosticRequest.getXMLDocument(), diagnosticRequest.getDiagnostics());
-				Diagnostic diag = internalValidateConfiguration(childDiagnosticReq, parameters);
-				if (diag != null) {
-					diagnosticRequest.getDiagnostics().add(diag);
-				}
+				DiagnosticRequest childDiagnosticReq = new DiagnosticRequest(childNode, diagnosticRequest);
+				internalValidateConfiguration(childDiagnosticReq, parameters);
 			}
 		}
-		return null;
 	}
 
-	public Diagnostic validateGoal(DiagnosticRequest diagnosticRequest) {
+	public void validateGoal(DiagnosticRequest diagnosticRequest) {
```

Comment:
Functions (which don't have side effects) usually leads to easier maintenance than methods like this one that are built on side effects.
So I'd suggest the method becomes something like private Optional<Diagnostic> validateGoal(DOMNode requestNode, descriptor), so consumers could just invoke validateGoal(node, descriptor).ifPresent(request::addDiagnostic). This would also allow easier fine grain testing if necessary.